<launch>

  <!-- ROS-Pybullet interface -->
  <node pkg="ros_pybullet_interface" name="ros_pybullet_interface" type="ros_pybullet_interface_node.py" output="screen">
    <rosparam param="config" file="$(find ros_quat_ds)/configs/run_franka/config.yaml"/>
  </node>


  <!-- Static transforms -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="floor_tf" args="0 0 0 0 0 0 rpbi/world rpbi/floor"/>

  <!-- Specify origin for IK target-->
  <node pkg="tf2_ros" type="static_transform_publisher" name="origin_tf" args="0.6 0 0.17 0 0 3.14 rpbi/world origin"/>



  <!-- IK -->
  <!-- == -->

  <!-- Setup IK -->
  <!-- <node pkg="ik_ros" name="ik_setup" type="ik_setup_node.py" args="trac_ik" output="screen">
    <param name="parent_frame_id" type="str" value="rpbi/franka/panda_link0"/>
    <param name="child_frame_id" type="str" value="target"/>
    <param name="start_on_init" type="bool" value="true"/>
    <remap from="ik/setup/trac_ik/qinit" to="rpbi/franka/joint_states"/>
  </node> -->


  <node pkg="ik_ros" name="ik_setup" type="ik_setup_node.py" args="trac_ik" output="screen">
    <param name="parent_frame_id" type="str" value="rpbi/franka/panda_link0"/>
    <param name="child_frame_id" type="str" value="target"/>
    <param name="start_on_init" type="bool" value="false"/>
    <remap from="ik/setup/trac_ik/qinit" to="rpbi/franka/joint_states"/>
  </node>



  <!-- IK solver interface -->
  <node pkg="ik_ros" name="ik_node" type="ik_solver_node.py" args="trac_ik" output="screen">
    <param name="urdf_filename" type="string" value="{franka_panda_description}/robots/panda_arm.urdf"/>
    <param name="tip_link" type="string" value="panda_link8"/>
    <param name="base_link" type="string" value="panda_link0"/>
    <param name="start_on_init" type="bool" value="true"/>
    <remap from="joint_states/target" to="rpbi/franka/joint_states/target"/> 
  </node>



  

  <!-- Example node that generates target joint positions -->
  <node pkg="ros_quat_ds" name="run_franka_eff_state" type="run_franka_eff_state.py" args="franka" output="screen"/>

</launch>
